From 0a8ec09322a41b1e3f601339ced914955e81c4ae Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 24 Jul 2017 14:44:34 +0800
Subject: [PATCH 2212/5242] MLK-16075-17 gpu: imx: dpu: fetchdecode: Add
 helper fetchdecode_need_fetcheco()

commit  5945f5dda9f22a3960a48960f75636445113a5d1 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds helper fetchdecode_need_fetcheco() so that users may
check if a fetchdecode needs to use fetcheco for a specific pixel format.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-fetchdecode.c |   21 +++++++++++++++++++++
 include/video/dpu.h                   |    1 +
 2 files changed, 22 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-fetchdecode.c b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
index bf5bc0c..333adcf 100644
--- a/drivers/gpu/imx/dpu/dpu-fetchdecode.c
+++ b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
@@ -523,6 +523,27 @@ struct dpu_fetcheco *fetchdecode_get_fetcheco(struct dpu_fetchdecode *fd)
 }
 EXPORT_SYMBOL_GPL(fetchdecode_get_fetcheco);
 
+bool fetchdecode_need_fetcheco(struct dpu_fetchdecode *fd, u32 fmt)
+{
+	struct dpu_fetcheco *fe = fetchdecode_get_fetcheco(fd);
+
+	if (IS_ERR_OR_NULL(fe))
+		return false;
+
+	switch (fmt) {
+	case DRM_FORMAT_NV12:
+	case DRM_FORMAT_NV21:
+	case DRM_FORMAT_NV16:
+	case DRM_FORMAT_NV61:
+	case DRM_FORMAT_NV24:
+	case DRM_FORMAT_NV42:
+		return true;
+	}
+
+	return false;
+}
+EXPORT_SYMBOL_GPL(fetchdecode_need_fetcheco);
+
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd)
 {
 	struct dpu_soc *dpu = fd->dpu;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index d1c0949..d099e4f 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -497,6 +497,7 @@ void fetchdecode_yuv_constantcolor(struct dpu_fetchdecode *fd,
 int fetchdecode_fetchtype(struct dpu_fetchdecode *fd, fetchtype_t *type);
 shadow_load_req_t fetchdecode_to_shdldreq_t(struct dpu_fetchdecode *fd);
 u32 fetchdecode_get_vproc_mask(struct dpu_fetchdecode *fd);
+bool fetchdecode_need_fetcheco(struct dpu_fetchdecode *fd, u32 fmt);
 unsigned int fetchdecode_get_stream_id(struct dpu_fetchdecode *fd);
 void fetchdecode_set_stream_id(struct dpu_fetchdecode *fd, unsigned int id);
 struct dpu_fetchdecode *dpu_fd_get(struct dpu_soc *dpu, int id);
-- 
1.7.9.5

